### PR TITLE
remove alphabet checks for uniprot

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request removes the alphabet checks in `Bio/SeqIO/UniprotIO.py`, and updates `test_SeqIO.py` accordingly.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_SeqIO.py [hunk 1]
```diff
@@ -672,18 +666,7 @@ def perform_test(
                         record = SeqIO.read(h, t_format, given_alpha)
                     self.assertIsInstance(base_alpha, given_base.__class__)
                     self.assertEqual(base_alpha, given_base)
-            for given_alpha in bad:
-                # These should all fail...
-                with open(t_filename, mode) as h:
-                    records6 = SeqIO.parse(h, t_format, given_alpha)
```
Comment (peterjc) [inline Tests/test_SeqIO.py:678]: I'm thinking that the SeqIO etc high level functions which take an alphabet will instead take the molecule type (perhaps under the same argument name), so this test would make sense with only minor edits.

File: Bio/SeqIO/UniprotIO.py [hunk 2]
```diff
@@ -44,12 +44,6 @@ def UniprotIterator(
     return_raw_comments = True --> comment fields are returned as complete XML to allow further processing
     skip_parsing_errors = True --> if parsing errors are found, skip to next entry
     """
-    if isinstance(alphabet, Alphabet.NucleotideAlphabet):
```
Comment (peterjc) [inline Bio/SeqIO/UniprotIO.py:47]: I think this should become something like ``"if alphabet and "protein" not in alphabet":`` where the default argument changes to ``alphabet="protein"`` (although we might want to change the argument name here and elsewhere in SeqIO/AlignIO etc)?

**Comment by peterjc** [comment]
On reflection, we can probably drop the alphabet argument to all the format specific parsers, and replace central old SeqIO ``_force_alphabet`` function with code to apply the molecule type to the SeqRecord before returning it to the caller.

https://github.com/biopython/biopython/blob/biopython-177/Bio/SeqIO/__init__.py#L645

File: Bio/SeqIO/UniprotIO.py (hunk 2 above)
Comment (peterjc) [reply Bio/SeqIO/UniprotIO.py:47]: On second thoughts, we can probably record/verify the molecule type centrally centrally in ``Bio.SeqIO.parse(...)``

**Comment by mdehoon** [comment]
@peterjc 
> On reflection, we can probably drop the alphabet argument to all the format specific parsers, and replace central old SeqIO _force_alphabet function with code to apply the molecule type to the SeqRecord before returning it to the caller.

Or we leave it to the user to add the molecule type after `SeqIO.parse` returns. Most users won't need to set the molecule type anyway. If they do, then it's trivial to set the molecule type once they start iterating over the records.

**Comment by peterjc** [comment]
Hmm. The ``Bio.SeqIO.convert(...)`` at least would need to set molecule type for some jobs.

**Comment by mdehoon** [comment]
@peterjc  For the main file formats, it only matters for Fasta-to-GenBank conversions, which probably few people do, and `Bio.SeqIO.convert` doesn't speed up the process there anyway.
This would work without having to iterate twice:
```
>>> records = SeqIO.parse(input_filename, 'fasta')
>>> def f(records):
...     for record in records:
...         record.annotations['molecule_type'] = 'RNA'
...         yield record
...
>>> SeqIO.write(f(records), output_filename, 'genbank')

**Comment by peterjc** [comment]
Yes, that is essentially what I have in mind for ``SeqIO.parse(input_filename, "fasta", mol_type="RNA")`` or reusing the existing ``alphabet=...`` argument.

**Comment by mdehoon** [comment]
@peterjc  My suggestion was to let the user take care of the molecule type, instead of adding this to `SeqIO.parse`.

**Pull request merged**